Read More functionality

adianasct opened the pull request:
Add collapsed character limit feature to Message and MessageBody components

- Introduced a new `collapsedCharacterLimit` prop to the Message and MessageBody components to control the maximum number of characters displayed before truncation.
- Implemented logic to handle text truncation and a "Read more" link for expanded view in MessageBody.
- Updated MessageList to pass the collapsed character limit down to Message components, ensuring consistent behavior across the application.

adianasct committed:
Add collapsed character limit feature to Message and MessageBody components

- Introduced a new `collapsedCharacterLimit` prop to the Message and MessageBody components to control the maximum number of characters displayed before truncation.
- Implemented logic to handle text truncation and a "Read more" link for expanded view in MessageBody.
- Updated MessageList to pass the collapsed character limit down to Message components, ensuring consistent behavior across the application.

karensct commented on src/components/Message/MessageBody/index.tsx:
```diff
@@ -319,6 +321,32 @@ const MessageBody = ({
+    if (collapsedCharacterLimit !== undefined && collapsedCharacterLimit !== null) {
+      const limit =
+        typeof collapsedCharacterLimit === 'number' ? collapsedCharacterLimit : Number(collapsedCharacterLimit)
+      return limit > 0 && !isNaN(limit) ? limit : 700
```
I told to add it 768

karensct commented on src/components/Message/MessageBody/index.tsx:
```diff
@@ -319,6 +321,32 @@ const MessageBody = ({
+  const characterLimit = useMemo(() => {
+    if (collapsedCharacterLimit !== undefined && collapsedCharacterLimit !== null) {
+      const limit =
+        typeof collapsedCharacterLimit === 'number' ? collapsedCharacterLimit : Number(collapsedCharacterLimit)
```
can character limit be 0?

karensct commented on src/components/Message/MessageBody/index.tsx:
```diff
@@ -1198,3 +1233,16 @@ const FrequentlyEmojisContainer = styled.div<{ rtlDirection?: boolean }>`
   z-index: 99;
 `
+
+const ReadMoreLink = styled.span<{ accentColor: string }>`
```
at all I don't se some animated opens view, I am sure it opens so ugly

karensct commented on src/components/Messages/MessageList/index.tsx:
```diff
@@ -479,7 +480,8 @@ const MessageList: React.FC<MessagesProps> = ({
   showInfoMessageProps = {},
-  ogMetadataProps
+  ogMetadataProps,
+  collapsedCharacterLimit = 700
```
you added default value it means it always will work, as I know you decided doesn't have it for default

adianasct committed:
Update Message components to handle text truncation and clipboard copying

- Enhanced the `handleCopyMessage` function in the Message component to copy truncated text based on a character limit.
- Adjusted the character limit default in MessageBody to 768 and updated logic for determining truncation.
- Added onCopy event handling in MessageBody to ensure the correct text is copied when the message is truncated.

adianasct committed:
.

adianasct committed:
Refine character limit handling in Message components

- Updated the logic in MessageBody to return undefined for invalid character limits (NaN or less than or equal to 0).
- Added `collapsedCharacterLimit` prop to MessagesContainer to ensure it is passed down to MessageBody, maintaining consistent truncation behavior.

adianasct committed:
..

adianasct committed:
Refactor clipboard copy logic in Message component

- Simplified the `handleCopyMessage` function to directly copy the message body if it is a valid string, removing unnecessary character limit checks and truncation logic.

karensct committed:
Update CarouselItem styles in sliderPopup component

- Removed the max-width restriction for images and videos, allowing them to utilize full width.
- Ensured consistent height and width settings for media elements to improve layout responsiveness.

karensct committed:
Enhance message copying functionality in Message component

- Updated the `handleCopyMessage` function to utilize the new `MessageTextFormat` utility for formatting message text before copying.
- Introduced `extractTextFromReactElement` to handle React elements and extract text content, ensuring accurate copying of formatted messages.
- Integrated `getShowOnlyContactUsers` to customize the text extraction based on contact visibility.

adianasct committed:
Implement message display logic for truncation in MessageBody component

- Added a new `displayMessage` useMemo hook to filter message body attributes based on character limit when truncation is enabled.
- Updated the rendering logic to use `displayMessage` instead of the original message, ensuring correct display of truncated messages.

adianasct committed:
..

adianasct committed:
.

karensct commented on src/components/Message/MessageBody/index.tsx:
```diff
@@ -319,6 +321,58 @@ const MessageBody = ({
+          setTextHeight(height)
+        }
+      })
+    } else if (!shouldTruncate) {
```
add checking for textHeight !== 'auto' then use setTextHeight('auto')

adianasct committed:
Refine message body truncation logic in MessageBody component

- Simplified the condition checks for message body validity in the `shouldTruncate` and `displayText` useMemo hooks, removing unnecessary type checks.
- Ensured that the truncation logic correctly handles cases where the message body is undefined.

karensct commented on src/components/Message/MessageBody/index.tsx:
```diff
@@ -749,22 +803,31 @@ const MessageBody = ({
-              onInviteLinkClick
-            })}
-          </span>
+          <React.Fragment>
```
why have you added react frgment ?

adianasct committed:
.

adianasct committed:
.

adianasct committed:
Update MessageList component to support character limit and refine truncation logic

- Added `collapsedCharacterLimit` prop to `MessageList` for better control over message display.
- Adjusted truncation logic in `MessageBody` to ensure accurate filtering of body attributes based on character limit.

karensct committed:
Enhance message truncation and formatting in MessageBody component

- Introduced `trimReactMessage` utility to improve text truncation logic, ensuring accurate handling of React nodes and mentions.
- Updated `MessageBody` to utilize the new `messageTextTrimmed` for rendering, simplifying the display logic and improving performance.
- Refined height adjustment logic based on truncation state, ensuring smoother transitions when expanding or collapsing message content.

karensct committed:
Merge branch 'fix/disappearing-messages-hiden-issue' of https://github.com/sceyt/sceyt-chat-react-uikit into feat/read-more-functionality
